clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1205,12 +1212,15 @@ public void assign(Collection<TopicPartition> partitions) {
                 throw new IllegalStateException("Consumer is not subscribed to any topics or assigned any partitions");
             }
 
+            this.kafkaConsumerMetrics.recordPollStart(time.milliseconds());
```

Comment:
I think we can change these calls to use `timer.currentTimeMs()` so that we do not add any additional calls to system time. These have been a performance issue in the past.